src/data/battler-tags.ts

```diff
@@ -2671,6 +2685,24 @@ export class SyrupBombTag extends BattlerTag {
   }
 }
 
+export class TelekinesisTag extends BattlerTag {
+  constructor(sourceMove: Moves) {
+    super(BattlerTagType.TELEKINESIS, [ BattlerTagLapseType.PRE_MOVE, BattlerTagLapseType.AFTER_MOVE ], 3, sourceMove, undefined, true);
+  }
+
+  override onAdd(pokemon: Pokemon)  {
+    pokemon.scene.queueMessage(i18next.t("battlerTags:telekinesisOnAdd", { pokemonNameWithAffix: getPokemonNameWithAffix(pokemon) }));
+    pokemon.addTag(BattlerTagType.TELEKINESIS, 3, this.sourceMove);
+    pokemon.addTag(BattlerTagType.FLOATING, 3, this.sourceMove);
+  }
+
+  override onRemove(pokemon: Pokemon) {
+    if (pokemon.getTag(BattlerTagType.FLOATING)) {
+      pokemon.removeTag(BattlerTagType.FLOATING);
+    }
```

Comment:
```suggestion
    pokemon.removeTag(BattlerTagType.FLOATING);
```
Slight optimization; `removeTag` just doesn't do anything if a matching tag isn't found.